docs/reference/r-listeners.md

```diff
@@ -4,22 +4,29 @@ Charmed Kafka comes with a set of listeners that can be enabled to allow for
-| SASL_INTERNAL | `cluster`                              | SASL_PLAINTEXT | `19092` | internal |
-| SASL_INTERNAL | `cluster` + `certificates`             | SASL_SSL       | `19093` | internal |
-| SASL_EXTERNAL | `kafka-client`                         | SASL_PLAINTEXT | `9092`  | external |
-| SASL_EXTERNAL | `kafka-client` + `certificates`        | SASL_SSL       | `9093`  | external |
-| SSL_EXTERNAL  | `trusted-certificate` + `certificates` | SSL            | `9094`  | external |
-| SSL_EXTERNAL  | `trusted-ca` + `certificates`          | SSL            | `9094`  | external |
+| Security protocol | Authentication mechanism | Driving endpoints                                      | Port    | Scope    | Listener name                         |
+|-------------------|--------------------------|--------------------------------------------------------|---------|----------|---------------------------------------|
+| SASL_PLAINTEXT    | SCRAM-SHA-512            | `cluster`                                              | `19092` | INTERNAL | INTERNAL_SASL_PLAINTEXT_SCRAM_SHA_512 |
+| SASL_SSL          | SCRAM-SHA-512            | `cluster` + `certificates`                             | `19093` | INTERNAL | INTERNAL_SASL_SSL_SCRAM_SHA_512       |
+| SASL_PLAINTEXT    | SCRAM-SHA-512            | `kafka-client`                                         | `9092`  | CLIENT   | CLIENT_SASL_PLAINTEXT_SCRAM_SHA_512   |
+| SASL_SSL          | SCRAM-SHA-512            | `kafka-client` + `certificates`                        | `9093`  | CLIENT   | CLIENT_SASL_SSL_SCRAM_SHA_512         |
+| SSL               | SSL                      | (`trusted-certificate`\|`trusted-ca`) + `certificates` | `9094`  | CLIENT   | CLIENT_SSL_SSL                        |
+| SASL_PLAINTEXT    | OAUTHBEARER              | `kafka-client` + `oauth`                               | `9095`  | CLIENT   | CLIENT_SASL_PLAINTEXT_OAUTHBEARER     |
+| SASL_SSL          | OAUTHBEARER              | `kafka-client` + `oauth` + `certificates`              | `9096`  | CLIENT   | CLIENT_SASL_SSL_OAUTHBEARER           |
+| SASL_PLAINTEXT    | SCRAM-SHA-512            | `kafka-client`                                         | `29092` | EXTERNAL | EXTERNAL_SASL_PLAINTEXT_SCRAM_SHA_512 |
+| SASL_SSL          | SCRAM-SHA-512            | `kafka-client` + `certificates`                        | `29093` | EXTERNAL | EXTERNAL_SASL_SSL_SCRAM_SHA_512       |
+| SSL               | SSL                      | (`trusted-certificate`\|`trusted-ca`) + `certificates` | `29094` | EXTERNAL | EXTERNAL_SSL_SSL                      |
+| SASL_PLAINTEXT    | OAUTHBEARER              | `kafka-client` + `oauth`                               | `29095` | EXTERNAL | EXTERNAL_SASL_PLAINTEXT_OAUTHBEARER   |
+| SASL_SSL          | OAUTHBEARER              | `kafka-client` + `oauth` + `certificates`              | `29096` | EXTERNAL | EXTERNAL_SASL_SSL_OAUTHBEARER         |
```

Comment:
**praise**: That's a really comprehensive resource to go back to

Reply:
*praise* I agree! Well done!